### PR TITLE
Updating specs for BUU (feature parity)

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

- Closes #12519 

Verifies the tests regarding the scenarios below (-> pointing to the relevant test cases in `./spec/system/admin/products_v3/products_spec.rb`):

```
As an Administrator
  I want to be able to manage products in bulk
  listing products:
```

- [x] displays a list of products -> `it "displays a list of products" do`
- [x] displays a message when number of products is zero -> `describe "with no products" do`
- [x] displays a select box for suppliers, with the appropriate supplier selected -> `it "can search for and update a product" do`
- [x] displays an on hand count in a span for each product -> `it "displays an on hand count in a span for each product" do`
- [x] displays 'on demand' for any variant that is available on demand -> `it "displays an on hand count in a span for each product" do`
- [x] displays a select box for the unit of measure for the product's variants -> `describe "Changing unit scale" do`
- [x] displays a text field for the item name when unit is set to 'Items' -> `it "saves a custom item unit name" do`

Also verified that:

- [x] creating a new product -> is covered in `./spec/system/admin/products_spec.rb`, and in the test it `"redirects to the New Product page" do`
- [x] creating new variants -> `describe "adding variants" do`
- [x] updating product attributes -> `describe "updating" do` (covers both changes on products and variants)
- [x] using edit buttons -> `describe "edit" do`
- [x] "using clone buttons" -> `describe "clone" do`
- [x] "using filtering controls" -> `describe "search" do`
- [x] as an enterprise manager -> `context "as an enterprise manager" do`
- [x] Updating product image -> `describe "edit image" do`
- [ ]  "using column display dropdown" -> `describe "using column display dropdown" do` -> "Pending implementation, issue #11055"

In this PR, test cases are moved (copied and then deleted) from:  
- `./spec/system/admin/products_spec.rb`

And copied updated from:
- `./spec/system/admin/bulk_product_update_spec.rb`

to:
- `./spec/system/admin/products_v3/products_spec.rb`

#### What should we test?

- green build

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: spec/system/admin/products_v3/products_spec.rb [hunk 1]
```diff
@@ -1336,6 +1336,38 @@
...
+                page.find(delete_option_selector).click
+              end
+
+              delete_button_selector = "input[type=button][value='Delete variant']"
+              within modal_selector do
+                page.find(delete_button_selector).click
+              end
+
+              expect(page).not_to have_selector(modal_selector)
+              expect(page).not_to have_selector(variant_selector)
```
Comment (rioug) [inline spec/system/admin/products_v3/products_spec.rb:1370]: Personally I don't like having `expect` in a before block, I would assume that deleting the variant is covered somewhere else so the `expect` are not necessary. If not I would just move this to the `it` block, we only have on `it` in this `context` to I think it's fine to have the set up in the `it` block.

File: spec/system/admin/products_v3/products_spec.rb [hunk 2]
```diff
@@ -1381,6 +1381,46 @@
...
+                                       permissions_list: [:manage_products])
+    end
+
+    before do
+      @enterprise_user = create(:user)
+      @enterprise_user.enterprise_roles.build(enterprise: supplier_managed1).save
+      @enterprise_user.enterprise_roles.build(enterprise: supplier_managed2).save
+      @enterprise_user.enterprise_roles.build(enterprise: distributor_managed).save
+
+      login_as @enterprise_user
```
Comment (rioug) [inline spec/system/admin/products_v3/products_spec.rb:1411]: You can just use `enterprise_user`, not need to create an instance variable here.

File: spec/system/admin/products_v3/products_spec.rb (hunk 1 above)
Comment (filipefurtad0) [reply spec/system/admin/products_v3/products_spec.rb:1370]: Makes sense - indeed, we do have the Delete scenario covered in another section of this spec; no point in testing it again - removed these two (negative) assertions.

File: spec/system/admin/products_spec.rb [hunk 3]
```diff
@@ -256,6 +256,81 @@
     end
   end
 
+  describe "deleting" do
+    let!(:product1) { create(:simple_product, name: 'a product to keep', supplier: @supplier) }
```
Comment (filipefurtad0) [inline spec/system/admin/products_spec.rb]: Info for reviewers: While in the middle of this PR, I've realized, it is perhaps not such a good idea to remove tests from the legacy design just yet, as we still have it in production. Hence added this commit to revert those changes (apologies for the detour!)

**Comment by filipefurtad0** [comment]
I'm not sure the build is green here... I think there is something going on with GH Actions. I'll move back to In progress and then to Code Review, if the situation resolves in the meantime, and the build here is indeed green.

**Review by mkllnk — CHANGES_REQUESTED** [review]

File: spec/system/admin/products_v3/products_spec.rb [hunk 4]
```diff
@@ -1446,6 +1446,34 @@
 
       expect(page).to have_selector row_containing_name(product_supplied_inactive.name)
     end
+
+    it "allows me to create a product" do
```
Comment (dacook) [inline spec/system/admin/products_v3/products_spec.rb:1449]: I thought this was already covered by "creating a new product" on line 626?

File: spec/system/admin/products_v3/products_spec.rb [hunk 5]
```diff
@@ -1419,23 +1419,22 @@
...
       visit spree.admin_products_path
-
       within row_containing_name(product_supplied.name) do
-        page.has_select?(
-          '_products_0_supplier_id',
-          options: [
-            supplier_managed1.name, supplier_managed2.name, supplier_permitted.name
-          ], selected: supplier_managed1.name
-        )
+        expect(page.has_select?(
```
Comment (dacook) [inline spec/system/admin/products_v3/products_spec.rb:1423]: This is fine, but FYI you can use the equivalent form:

`expect(page).to have_select(`

(RSpec magically handles a `have_*` matcher for any `has_*?` methods)

**Review by dacook — APPROVED** [review]
Great, thanks for working through this!

I think maybe one test could be removed, let me know what you think. Otherwise all good 👍

File: spec/system/admin/products_v3/products_spec.rb [hunk 6]
```diff
@@ -1416,6 +1416,28 @@
       expect(page).to have_selector row_containing_name(product_supplied_permitted.name)
       expect(page).not_to have_selector row_containing_name(product_not_supplied.name)
     end
+
+    it "shows only suppliers that I manage or have permission to" do
+      visit spree.admin_products_path
+
+      within row_containing_name(product_supplied.name) do
+        page.has_select?(
```
[the remark this answers is not in the document]
Comment (filipefurtad0) [reply spec/system/admin/products_v3/products_spec.rb]: Nice catch @mkllnk . I've addressed this issue and combined has_select? with the `expect`.

File: spec/system/admin/products_v3/products_spec.rb (hunk 5 above)
Comment (filipefurtad0) [reply spec/system/admin/products_v3/products_spec.rb]: Thanks @dacook , it was working but the `have_select()` syntax is really much simpler. Squashed it to the previous commit.

File: spec/system/admin/products_v3/products_spec.rb (hunk 4 above)
Comment (filipefurtad0) [reply spec/system/admin/products_v3/products_spec.rb:1449]: Yes, you're right. I've added it anyway because the context was on enterprise permissions. But indeed it's doing the same as the test on line 626 (I've opted to only test the redirect here) and - more thoroughly - the test on `./spec/system/admin/products_spec.rb` ([here](https://github.com/openfoodfoundation/openfoodnetwork/blob/e22bec014bd0318131dba17461b18a46bde0ab0b/spec/system/admin/products_spec.rb#L29)). Removing it :+1:

**Review by mkllnk — APPROVED** [review]

**Review by rioug — APPROVED** [review]
Looks good :+1:
